src/components/instructor_files/instructor_files.vue

```diff
@@ -0,0 +1,214 @@
+    collapsed = false;
+    num_files_currently_viewing = 0;
+
+    @Prop({required: true, type: Project})
+    project!: Project;
+
+    async created() {
+      InstructorFile.subscribe(this);
+      this.instructor_files = await InstructorFile.get_all_from_project(this.project.pk);
+      this.sort_files();
+    }
+
+    destroyed() {
+      InstructorFile.unsubscribe(this);
+    }
+
+    sort_files() {
+      this.instructor_files.sort(
+        (file_a: InstructorFile, file_b: InstructorFile) => {
+        if (file_a.name <= file_b.name) {
+          return -1;
+        }
+        else {
+          return 1;
+        }
+      });
+    }
+
+    async view_file(file: InstructorFile) {
+      (<MultiFileViewer> this.$refs.instructor_files_viewer).add_to_viewing(
+        file.name,  () => file.get_content(), file.pk
+      );
+    }
+
+    async add_instructor_files(files: File[]) {
+      for (let file of files) {
+        let file_already_exists = array_has_unique(
+          this.instructor_files,
+          file.name,
+          (file_a: InstructorFile, file_name_to_add: string) =>
+            file_name_equal(file_a.name, file_name_to_add)
+        );
+
+        if (file_already_exists) {
+          let file_to_update = array_get_unique(
+            this.instructor_files,
+            file.name,
+            (file_a: InstructorFile, file_name_to_add: string) =>
+              file_name_equal(file_a.name, file_name_to_add)
+          );
+          await file_to_update.set_content(file);
+        }
+        else {
+            let file_to_add = await InstructorFile.create(this.project.pk, file.name, file);
+            this.instructor_files.push(file_to_add);
+        }
+      }
+      (<FileUpload> this.$refs.instructor_files_upload).clear_files();
+      this.sort_files();
+    }
+
+    update_instructor_file_content_changed(instructor_file: InstructorFile,
+                                           file_content: string) {
+      (<MultiFileViewer> this.$refs.instructor_files_viewer).update_contents_by_name(
+        instructor_file.name, file_content
+      );
+    }
+
+    update_instructor_file_created(instructor_file: InstructorFile) { }
+
+    update_instructor_file_deleted(instructor_file: InstructorFile) {
+      array_remove_unique(this.instructor_files, instructor_file.pk, (file, pk) => file.pk === pk);
+      (<MultiFileViewer> this.$refs.instructor_files_viewer).remove_by_name(instructor_file.name);
+    }
+
+    update_instructor_file_renamed(instructor_file: InstructorFile) {
+      let index = this.instructor_files.findIndex((file) => file.pk === instructor_file.pk);
+      Vue.set(this.instructor_files, index, instructor_file);
+      (<MultiFileViewer> this.$refs.instructor_files_viewer).rename_file(
+        instructor_file.pk, instructor_file.name
+      );
+      this.sort_files();
+    }
+  }
+
+  function file_name_equal(file_name_a: string, file_name_b: string) {
```

Comment:
Is there a particular need for this function, or can we just inline the === check?